### DISCUSSION TITLE
How to update to latest version?

### DISCUSSION THREAD

**Discussion by angyen** [microrealestate/microrealestate#141] [Q&A] [answered]
Dear developers,

I am new in this and have successfully build and run mre after multiple tries 3 weeks ago.

May I know what is the correct step to update the app?

Does updating the app affect the current data which are already there?

I am hoping to be able to update the app without losing data.

Thank you in advance.

**Comment by camelaissani** [comment, marked as answer]
1. Save your data for more safety
In the folder of the repository run this command when the app is running
```sh
./mre dumpdb
```
This will create a timestamp dump file in the `backup` folder

Then copy the dump file and .env file in a safe directory (outside of the repository folder)

2. Update the app
Stop the application
```sh
./mre stop
```
Pull the latest source files
```sh
git pull
```
3. Build the docker images
```sh
./mre build
```
4. Start the app
```sh
./mre start
```
The app should be updated without any data loss

> In case of loss of data:
> Stop the app
> ```sh
> ./mre stop
> ```
> Copy the copied .env file (file from step1) in the root folder of the app.
> Copy the dump file (file from step1) in the backup directory of the app.
> Restart the app
> ```sh
> ./mre start
> ```
> The app will be started by using the environment variables from your .env file.
> Restore the data from the dump file
> ```sh
> ./mre restoredb
> ```
> Select your dump file when the command ask to select a backup
> Then the app should have data restored 
>

**Reply by angyen** [reply]
Thank you for the detail guide.

I am able to update the app without data loss.

I backup the data with `./mre dumpdb` instead as `./mre dump` doesn't work.

**Reply by angyen** [reply]
I am sorry but to ask again.
Since the new method of installation doesn't involve 'build' anymore, are we safe to update by just using the 'curl' and 'tar' command?

**Reply by camelaissani** [reply]
Pretty much the same procedure:

1. Save your data for more safety
In the folder of the repository run this command when the app is running
```sh
./mre dumpdb
```
This will create a timestamp dump file in the `backup` folder

Then copy the dump file and .env file in a safe directory (outside of the repository folder)

2. Stop the application
```sh
./mre stop
```

3. Get the latest version

Follow this [guide](https://github.com/microrealestate/microrealestate?tab=readme-ov-file#1-install-the-application) but do not start the application.
The destination directory for downloading the application must not be a clone of the repository. 

4. Start the app
Copy the copied .env file (file from step1) in the root folder of the app.
Copy the dump file (file from step1) in the backup directory of the app.
Restart the app
```sh
./mre start
```
The app will be started by using the environment variables from your .env file.
Restore the data from the dump file
```sh
./mre restoredb
```
Select your dump file when the command ask to select a backup
